app/code/Magento/CatalogInventory/Model/Quote/Item/QuantityValidator/Initializer/StockItem.php

```diff
@@ -92,9 +92,9 @@ public function initialize(
 
         $result = $this->stockState->checkQuoteItemQty(
             $product->getId(),
-            $rowQty,
             $qtyForCheck,
-            $qty,
+            $qty,         
+            (is_object($quoteItem->getStockStateResult()) ? $quoteItem->getStockStateResult()->getOrigQty() : $quoteItem->getQty()),
```

Comment:
@maghamed Could you provide your feedback
This part looks very strange

Reply:
I am not sure that created https://github.com/magento/graphql-ce/issues/477 issue is a real issue.

I described a situation in the issue ticket . However, I can add explanations here:

In the first Tomash query he added to Cart `1` x `Product A`.

In the second one, Tomash added `0.5` x `Product A` to the same shopping cart. 

So, in the background process Tomash added `1.5` x `Product A` into shopping cart because `Product A` with quantity `1` already exists in the cart + a new quantity `0.5`. 

In the result, Tomash added to cart `1.5` x `Product A`. 

`Product A` does not support decimal product qty that is why `1.5` has been converted into `1`.

Thank you!